client/question/question.js

```diff
@@ -20,6 +20,71 @@ window.tie = angular.module('tie',
   ['ui.codemirror', 'tieConfig', 'tieData', 'ngCookies']);
 
 /**
+ * String used to label test suites that contain sample input and output
+ * that are presented to the user.
+ *
+ * @type {string}
+ */
+tie.constant('SAMPLE_INPUT', 'SAMPLE_INPUT'); 
+
+/**
+ * Correctness state where the test case input has been displayed to the user.
+ *
+ * @type {string}
+ */
+tie.constant('CORRECTNESS_TEST_INPUT_DISPLAYED',
+  'CORRECTNESS_TEST_INPUT_DISPLAYED');
+
+/**
+ * Correctness state where the expeced output has been displayed to the user.
+ *
+ * @type {string}
+ */
+tie.constant('CORRECTNESS_TEST_EXPECTED_OUTPUT_DISPLAYED', 
+  'CORRECTNESS_TEST_EXPECTED_OUTPUT_DISPLAYED');
+
+/**
+ * Correctness state where the observed output has been displayed to the user.
+ *
+ * @type {string}
+ */
+tie.constant('CORRECTNESS_TEST_OBSERVED_OUTPUT_DISPLAYED',
+  'CORRECTNESS_TEST_OBSERVED_OUTPUT_DISPLAYED');
+
+/**
+ * Object containing variations on feedback text for different scenarios.
+ *
+ * @type {Object.<string, Array.<string>>}
+ */
+tie.constant('CORRECTNESS_FEEDBACK_TEXT', {
+  'INPUT_TO_TRY': [
```

Comment:
These keys could match the values in the constants above?

E.g. above, define: tie.constant('CORRECTNESS_STATE_KEY_INPUT_DISPLAYED', 'INPUT_DISPLAYED');

and here:

```
tie.constant('CORRECTNESS_FEEDBACK_TEXT', {
  INPUT_DISPLAYED: [...]
});
```